### PR TITLE
First version of photon filter

### PR DESCRIPTION
**Pull request by peressounko**
First draft of EM filter for photons etc.
Content still to be discussed

### CODE REVIEW

**Comment by peressounko** [comment]
Dear @mpuccio, could you please have a look at the draft? I have some doubts how to organize code. The problem is: in PHOS/ECMAL/CPV we have cells/clusters readout per BC. In most cases there is one-to-one correspondence between BC and Collision but in the case of ambiguity we would like to keep assignment to BC. Alternatively, we can keep assignment to BC for all cases and solve ambiguity in analysis. Therefore, for PHOS it would be convenient not to filter clusters belonging to current Collision and fill tag row (as I see for other tasks) but to fill all rows for a full TF.  Then I am a bit confused, how final table should be organized and filled.
Could you have a look and let me know, if such design is appropriate at al and how to fill final table (what should be the index column)?
Thanks, Dmitri

**Comment by mpuccio** [comment]
Hi @peressounko, as I mentioned via email, if you select a collision and the association to BC is ambiguous, we will store all the BCs compatible with the filtered collision id. So you don't need to do disambiguation in your task.
By the way, currently your task does not compile in o2, see the errors in the CI.

**Comment by peressounko** [comment]
Dear @mpuccio, could you please have a look at this PR now - it passes checks.

File: EventFiltering/PWGEM/photonFilter.cxx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  Configurable<float> ePhot{"ePhot", 2., "Minimal photon energy (GeV)"};
+  Configurable<float> eEl{"eEl", 2., "Minimal electron energy (GeV)"};
+  Configurable<float> ePair{"ePair", 0.5, "Minimal photon pair mass (GeV)"};
+
+  HistogramRegistry mHistManager{"events", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+
+  void init(o2::framework::InitContext&)
+  {
+    auto scalers{std::get<std::shared_ptr<TH1>>(mHistManager.add("fProcessedEvents", "Number of filtered events", HistType::kTH1F, {{10, -0.5, 9.5}}))};
+    scalers->GetXaxis()->SetBinLabel(0, "Processed events");
```
Comment (mpuccio) [inline EventFiltering/PWGEM/photonFilter.cxx:46]: SetBinLabel starts from 1

File: EventFiltering/PWGEM/photonFilter.cxx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        }
+        if (keepEvent[kPair]) {
+          mHistManager.fill(HIST("fProcessedEvents"), 3.);
+        }
+        if (keepEvent[kNbar]) {
+          mHistManager.fill(HIST("fProcessedEvents"), 4.);
+        }
+
+        // fill
+        tags(keepEvent[kPhot], keepEvent[kEl], keepEvent[kPair], keepEvent[kNbar]);
```
Comment (mpuccio) [inline EventFiltering/PWGEM/photonFilter.cxx:100]: You should fill only once per collision

File: EventFiltering/PWGEM/photonFilter.cxx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      mHistManager.fill(HIST("fProcessedEvents"), 2.);
+    }
+    if (keepEvent[kPair]) {
+      mHistManager.fill(HIST("fProcessedEvents"), 3.);
+    }
+    if (keepEvent[kNbar]) {
+      mHistManager.fill(HIST("fProcessedEvents"), 4.);
+    }
+    // fill
+    tags(keepEvent[kPhot], keepEvent[kEl], keepEvent[kPair], keepEvent[kNbar]);
```
Comment (mpuccio) [inline EventFiltering/PWGEM/photonFilter.cxx:157]: Again, you should have only one entry per collision

**Review by mpuccio — CHANGES_REQUESTED** [review]
Hi Dmitri, please implement the comments I sent you. Then I will merge this PR.

**Comment by mpuccio** [comment]
Hi @peressounko, is this ready to be reviewed again?

**Comment by peressounko** [comment]
Hi @mpuccio, yes: I implemented your suggestions and checks passed

**Comment by mpuccio** [comment]
Can you please rebase your branch? There is a spurious commit.

**Comment by mpuccio** [comment]
I still see a commit from Victor in your PR.

**Comment by peressounko** [comment]
dear @mpuccio, Victor modified file filterTables.h If I remove his commit, we will not be able to merge as there will be merge conflict.

**Comment by peressounko** [comment]
Dear @mpuccio now checks passed, no spurious commit, your suggestions implemented. Could you review?

**Review by mpuccio — APPROVED** [review]

**Pull request merged**